README.md
# Hackathon2022

# Team Members
1. Jose Martinez
1. Julio Sibrian
3. Johnathan Zemuduo
4. Sam Lefforge
5. Alexis Mercado
6. Steve Liang

# Project Description
Project Description:  A VS code extension that makes passive aggressive remarks about your bad stylistic coding choices. 



# PASyntax README

This is the README for your extension "PASyntax". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: enable/disable this extension
* `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

-----------------------------------------------------------------------------------------------------------

## Working with Markdown

**Note:** You can author your README using Visual Studio Code.  Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux)
* Toggle preview (`Shift+CMD+V` on macOS or `Shift+Ctrl+V` on Windows and Linux)
* Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (macOS) to see a list of Markdown snippets

### For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**


CHANGELOG.md
# Change Log

All notable changes to the "PASyntax" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

PASyntax/PASyntax/src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
// import { TypesInstaller } from './typesInstaller';

async function getDiagnostics(doc: vscode.TextDocument): Promise<vscode.Diagnostic[]> {
	const text = doc.getText();
	const diagnostics = new Array<vscode.Diagnostic>();
	let testing = false;

	const textArr: string[] = text.split(/\r\n|\n/);

	for (let i = 0; i < textArr.length; i++) {
		let found = -1;
		found = textArr[i].search("testing");
		if (found > 0) {
			var start = textArr[i].search("testing");
			var end = start + 7;
			// for (let j = 0; j < textArr[i].length; j++) {
			// 	if (textArr[i].charAt(j) == 'x') {
			// 		start = j;
			// 		end = j + 13;
			// 		break;
			// 	}
			// }
			diagnostics.push({
				severity: vscode.DiagnosticSeverity.Warning,
				message: `testing squiggily`,
				code: 'testing dont work',
				source: 'PASyntax',
				range: new vscode.Range(i, start, i, end)
			});
		}
		found = textArr[i].search("penis")
		if (found >= 0) {
			var start = textArr[i].search("penis");
			var end = start + 5;
			// for (let j = 0; j < textArr[i].length; j++) {
			// 	if (textArr[i].charAt(j) == 'x') {
			// 		start = j;
			// 		end = j + 13;
			// 		break;
			// 	}
			// }
			diagnostics.push({
				severity: vscode.DiagnosticSeverity.Information,
				message: `penis`,
				code: 'penis dont work',
				source: 'PASyntax',
				range: new vscode.Range(i, start, i, end)
			});
		}
	}

	return diagnostics;
}

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export async function activate(context: vscode.ExtensionContext) {
	const diagnosticCollection = vscode.languages.createDiagnosticCollection('types-installer');
	
	const handler = async (doc: vscode.TextDocument) => {
		if(!doc.fileName.endsWith('meow.java')) {
			console.log("left")
			return;
		}
	
		const diagnostics = await getDiagnostics(doc);
		diagnosticCollection.set(doc.uri, diagnostics);
	};
	
	const didOpen = vscode.workspace.onDidOpenTextDocument(doc => handler(doc));
	const didChange = vscode.workspace.onDidChangeTextDocument(e => handler(e.document));
	// const codeActionProvider = vscode.languages.registerCodeActionsProvider('java', new TypesInstaller(context));
	
	// If we have an activeTextEditor when we open the workspace, trigger the handler
	if (vscode.window.activeTextEditor) {
		await handler(vscode.window.activeTextEditor.document);
	}

	context.subscriptions.push(
		vscode.languages.registerHoverProvider("java", {
		  provideHover: (document, position) => {
			const {line} = position;
			const ur = "https://gray-wbrc-prod.cdn.arcpublishing.com/resizer/_auGpao1CPapEPc4UeaKOiKnZ5Y=/1200x1800/smart/filters:quality(85)/cloudfront-us-east-1.images.arcpublishing.com/gray/BV7IA24BUZBBRCKTJE5XP6KPPQ.jpg";

			const lineContent = document.lineAt(line).text;
			const regexp =
			  /test/;
			const res = lineContent.match(regexp);
			console.log("1", res);
			if (res === null) {
			  return;
			}
			const url = res[0];
			console.log("2", url);
			return new vscode.Hover(new vscode.MarkdownString(`![](${ur}|width=150,height=150)`));
		  },
		})
	  );
	
	// Push all of the disposables that should be cleaned up when the extension is disabled
	context.subscriptions.push(
		diagnosticCollection,
		didOpen,
		didChange,
		// codeActionProvider
		);

	
	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
	console.log('Congratulations, your extension "PASyntax" is now active!');

	// The command has been defined in the package.json file
	// Now provide the implementation of the command with registerCommand
	// The commandId parameter must match the command field in package.json
	let disposable = vscode.commands.registerCommand('PASyntax.helloWorld', () => {
		// The code you place here will be executed every time your command is executed
		// Display a message box to the user
		vscode.window.showInformationMessage('Meow Meow');
	});

	

	context.subscriptions.push(disposable);
}

// this method is called when your extension is deactivated
export function deactivate() {}





PASyntax/PASyntax/out/typesInstaller.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypesInstaller = void 0;
const vscode = require("vscode");
class TypesInstaller {
    constructor(context) {
        const command = vscode.commands.registerCommand('types-installer.installTypesModule', async (range) => {
            vscode.window.activeTextEditor?.document.save();
            const text = vscode.window.activeTextEditor?.document.getText(range);
            const useYarn = !!(await vscode.workspace.findFiles('yarn.lock'));
            const shellExec = useYarn
                ? new vscode.ShellExecution(`yarn add --dev @types/${text}`)
                : new vscode.ShellExecution(`npm i --save-dev @types/${text}`);
            vscode.tasks.executeTask(new vscode.Task({ type: 'typesinstaller' }, vscode.TaskScope.Workspace, 'TypesInstaller', 'Types Installer', shellExec, 'npm'));
        });
        context.subscriptions.push(command);
    }
    provideCodeActions(document, range, context, token) {
        // for each diagnostic entry that has the matching `code`, create a code action command
        return context.diagnostics
            .filter(diagnostic => diagnostic.code === 'no-types-detected')
            .map(diagnostic => this.createCommandCodeAction(diagnostic));
    }
    createCommandCodeAction(diagnostic) {
        const text = vscode.window.activeTextEditor?.document.getText(diagnostic.range);
        const action = new vscode.CodeAction(`Install @types/${text} module...`, vscode.CodeActionKind.QuickFix);
        action.diagnostics = [diagnostic];
        action.isPreferred = true;
        action.command = {
            command: 'types-installer.installTypesModule',
            title: 'Learn more about emojis',
            tooltip: 'This will open the unicode emoji page.',
            arguments: [diagnostic.range]
        };
        return action;
    }
}
exports.TypesInstaller = TypesInstaller;
//# sourceMappingURL=typesInstaller.js.map

PASyntax/PASyntax/src/typesInstaller.ts
import * as vscode from 'vscode';

export class TypesInstaller implements vscode.CodeActionProvider {
    constructor(context: vscode.ExtensionContext) {
        const command = vscode.commands.registerCommand('types-installer.installTypesModule', async (range: vscode.Range) => {
            vscode.window.activeTextEditor?.document.save();
            const text = vscode.window.activeTextEditor?.document.getText(range);
            const useYarn = !!(await vscode.workspace.findFiles('yarn.lock'));
            const shellExec = useYarn
                ? new vscode.ShellExecution(`yarn add --dev @types/${text}`)
                : new vscode.ShellExecution(`npm i --save-dev @types/${text}`);
    
            vscode.tasks.executeTask(
                new vscode.Task({ type: 'typesinstaller' },
                    vscode.TaskScope.Workspace,
                    'TypesInstaller',
                    'Types Installer',
                    shellExec,
                    'npm'));
        });
        context.subscriptions.push(command);
    }

    provideCodeActions(document: vscode.TextDocument, range: vscode.Range | vscode.Selection, context: vscode.CodeActionContext, token: vscode.CancellationToken): vscode.ProviderResult<(vscode.CodeAction | vscode.Command)[]> {
        // for each diagnostic entry that has the matching `code`, create a code action command
        return context.diagnostics
            .filter(diagnostic => diagnostic.code === 'no-types-detected')
            .map(diagnostic => this.createCommandCodeAction(diagnostic));
    }

    private createCommandCodeAction(diagnostic: vscode.Diagnostic): vscode.CodeAction {
        const text = vscode.window.activeTextEditor?.document.getText(diagnostic.range);
        const action = new vscode.CodeAction(`Install @types/${text} module...`, vscode.CodeActionKind.QuickFix);
        action.diagnostics = [ diagnostic ];
        action.isPreferred = true;
        action.command = {
            command: 'types-installer.installTypesModule',
            title: 'Learn more about emojis',
            tooltip: 'This will open the unicode emoji page.',
            arguments: [ diagnostic.range ] 
        };
        return action;
    }
}

PASyntax/PASyntax/out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
const vscode = require("vscode");
// import { TypesInstaller } from './typesInstaller';
async function getDiagnostics(doc) {
    const text = doc.getText();
    const diagnostics = new Array();
    let testing = false;
    const textArr = text.split(/\r\n|\n/);
    for (let i = 0; i < textArr.length; i++) {
        let found = -1;
        found = textArr[i].search("testing");
        if (found > 0) {
            var start = textArr[i].search("testing");
            var end = start + 7;
            // for (let j = 0; j < textArr[i].length; j++) {
            // 	if (textArr[i].charAt(j) == 'x') {
            // 		start = j;
            // 		end = j + 13;
            // 		break;
            // 	}
            // }
            diagnostics.push({
                severity: vscode.DiagnosticSeverity.Warning,
                message: `testing squiggily`,
                code: 'testing dont work',
                source: 'PASyntax',
                range: new vscode.Range(i, start, i, end)
            });
        }
        found = textArr[i].search("penis");
        if (found >= 0) {
            var start = textArr[i].search("penis");
            var end = start + 5;
            // for (let j = 0; j < textArr[i].length; j++) {
            // 	if (textArr[i].charAt(j) == 'x') {
            // 		start = j;
            // 		end = j + 13;
            // 		break;
            // 	}
            // }
            diagnostics.push({
                severity: vscode.DiagnosticSeverity.Information,
                message: `penis`,
                code: 'penis dont work',
                source: 'PASyntax',
                range: new vscode.Range(i, start, i, end)
            });
        }
    }
    return diagnostics;
}
// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
async function activate(context) {
    const diagnosticCollection = vscode.languages.createDiagnosticCollection('types-installer');
    const handler = async (doc) => {
        if (!doc.fileName.endsWith('meow.java')) {
            console.log("left");
            return;
        }
        const diagnostics = await getDiagnostics(doc);
        diagnosticCollection.set(doc.uri, diagnostics);
    };
    const didOpen = vscode.workspace.onDidOpenTextDocument(doc => handler(doc));
    const didChange = vscode.workspace.onDidChangeTextDocument(e => handler(e.document));
    // const codeActionProvider = vscode.languages.registerCodeActionsProvider('java', new TypesInstaller(context));
    // If we have an activeTextEditor when we open the workspace, trigger the handler
    if (vscode.window.activeTextEditor) {
        await handler(vscode.window.activeTextEditor.document);
    }
    context.subscriptions.push(vscode.languages.registerHoverProvider("java", {
        provideHover: (document, position) => {
            const { line } = position;
            const ur = "https://gray-wbrc-prod.cdn.arcpublishing.com/resizer/_auGpao1CPapEPc4UeaKOiKnZ5Y=/1200x1800/smart/filters:quality(85)/cloudfront-us-east-1.images.arcpublishing.com/gray/BV7IA24BUZBBRCKTJE5XP6KPPQ.jpg";
            const lineContent = document.lineAt(line).text;
            const regexp = /test/;
            const res = lineContent.match(regexp);
            console.log("1", res);
            if (res === null) {
                return;
            }
            const url = res[0];
            console.log("2", url);
            return new vscode.Hover(new vscode.MarkdownString(`![](${ur}|width=150,height=150)`));
        },
    }));
    // Push all of the disposables that should be cleaned up when the extension is disabled
    context.subscriptions.push(diagnosticCollection, didOpen, didChange);
    // Use the console to output diagnostic information (console.log) and errors (console.error)
    // This line of code will only be executed once when your extension is activated
    console.log('Congratulations, your extension "PASyntax" is now active!');
    // The command has been defined in the package.json file
    // Now provide the implementation of the command with registerCommand
    // The commandId parameter must match the command field in package.json
    let disposable = vscode.commands.registerCommand('PASyntax.helloWorld', () => {
        // The code you place here will be executed every time your command is executed
        // Display a message box to the user
        vscode.window.showInformationMessage('Meow Meow');
    });
    context.subscriptions.push(disposable);
}
exports.activate = activate;
// this method is called when your extension is deactivated
function deactivate() { }
exports.deactivate = deactivate;
//# sourceMappingURL=extension.js.map